src/test/lines.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {strMult} from "./utils";
+
+// Use classes here instead of interfaces because we want runtime type data
+export class Line { }
+export class CodeLine extends Line { constructor(public contents: string) { super(); } }
+export class MessageSubstitutionLine extends Line { constructor(public key: string, public message: string) { super(); } }
+
+export class ErrorLine extends Line { constructor(public startCol: number) { super(); } }
+export class MultilineErrorLine extends ErrorLine { constructor(startCol: number) { super(startCol); } }
+export class EndErrorLine extends ErrorLine {
+    constructor(startCol: number, public endCol: number, public message: string) { super(startCol); }
+}
+
+// example matches (between the quotes): 
+// "    ~~~~~~~~"
+const multilineErrorRegex = /^\s*(~+|~nil)$/;
+// "    ~~~~~~~~~   [some error message]"
+const endErrorRegex = /^\s*(~+|~nil)\s*\[(.+)\]\s*$/;
+// "[shortcut]: full messages goes here!!  "
+const messageSubstitutionRegex = /^\[([\w\-\_]+?)]: \s*(.+?)\s*$/;
+
+export const ZERO_LENGTH_ERROR = "~nil";
+
+/**
+ * Maps a line of text from a .linttest file to an appropriate Line object
+ */
+export function textToLineObj(text: string): Line {
```

Comment:
"obj" sounds a little clunky, how about we omit it? the return type tells you it's an object.